apps/fluent-tester/src/FluentTester/TestComponents/Notification/NotificationTest.tsx

```diff
@@ -9,128 +10,464 @@ const svgProps: SvgIconProps = {
+            variant={'primary'}
+            visible={visible}
+            action="Undo"
+            onPress={() => {
+              console.log('Notification tapped');
+            }}
+            onActionPress={() => {
+              console.log('Undo tapped');
+            }}
+          >
+            Mail Archived
+          </Notification>
+        </Animated.View>
+      )}
+    </View>
   );
 };
 
 const PrimaryTestWithTitleAndIcon: React.FunctionComponent = () => {
+  const [visible, setVisible] = React.useState(false);
+  const onButtonPress = () => setVisible(!visible);
+
+  const [hidden, setHidden] = React.useState<boolean>(!visible);
+  const height = React.useRef(new Animated.Value(0)).current;
+
+  React.useLayoutEffect(() => {
+    if (visible) {
+      setHidden(false);
+      Animated.timing(height, {
+        toValue: -50,
+        duration: 600,
+        useNativeDriver: true,
+        easing: Easing.elastic(1.5),
+      }).start();
+    } else {
+      Animated.timing(height, {
+        toValue: 0,
+        duration: 250,
+        useNativeDriver: true,
+      }).start(() => {
+        setHidden(true);
+      });
+    }
+  }, [visible, height]);
+
+  const animatedViewProps = {
+    transform: [{ translateY: height }],
+  };
+
   return (
-    <Notification
-      variant={'primary'}
-      icon={iconProps}
-      title="Kat's iPhoneX"
-      onPress={() => {
-        console.log('Notification tapped');
-      }}
-    >
-      Listen to Emails • 7 mins
-    </Notification>
+    <View>
+      <View style={{ marginBottom: 100 }}>
+        <Button onPress={onButtonPress} title={visible ? 'Hide' : 'Show'} />
+      </View>
+      {!hidden && (
+        <Animated.View style={[animatedViewProps]}>
+          <Notification
+            variant={'primary'}
+            visible={visible}
+            icon={iconProps}
+            title="Kat's iPhoneX"
+            onPress={() => {
+              console.log('Notification tapped');
+            }}
+          >
+            Listen to Emails • 7 mins
+          </Notification>
+        </Animated.View>
+      )}
+    </View>
   );
 };
 
 const NeutralTest: React.FunctionComponent = () => {
+  const [visible, setVisible] = React.useState(false);
+  const onButtonPress = () => setVisible(!visible);
+
+  const [hidden, setHidden] = React.useState<boolean>(!visible);
+  const height = React.useRef(new Animated.Value(0)).current;
+
+  React.useLayoutEffect(() => {
+    if (visible) {
+      setHidden(false);
+      Animated.timing(height, {
+        toValue: -50,
+        duration: 600,
+        useNativeDriver: true,
+        easing: Easing.elastic(1.5),
+      }).start();
+    } else {
+      Animated.timing(height, {
+        toValue: 0,
+        duration: 250,
+        useNativeDriver: true,
+      }).start(() => {
+        setHidden(true);
+      });
+    }
+  }, [visible, height]);
+
+  const animatedViewProps = {
+    transform: [{ translateY: height }],
+  };
```

Comment:
Any chance we can pull out this logic into something re-usable so we don't need to keep copy-pasting everywhere?

Reply:
An idea could be to combine all the tests into one test with a Picker to set the variant (similar to the pickers for Brand / Theme)

Reply:
I did attempt to do this but I believe it violates some hook rules (hooks should only be called at the top level). [https://reactjs.org/docs/hooks-rules.html](url)